add-rest-file_storage

CookChicker opened the pull request:
Rest file storage task

CookChicker committed:
Добавил путь для сохранения картинок итема.

CookChicker committed:
Рест контроллер для загрузки файлов в обе стороны.

CookChicker committed:
Для работы с .png файлами.

CookChicker committed:
Для работы с рест file storage.

CookChicker committed:
Минус костыли.

NikitaNesterenko commented on service-impl/src/main/java/jm/stockx/FileStorageServiceImpl.java:
```diff
@@ -0,0 +1,76 @@
+            return "Transferred a non-existent file.";
+        }
+
+        Item item = itemService.getItemById(id);
```
зачем тащить весь итем, чтоб сохранить лишь ссылку? просто запросом сохрани

CookChicker committed:
Теперь link добавляется через update запрос, не вытягивая итем.

CookChicker committed:
Для отлова ошибок API

CookChicker committed:
Обработал следующее:
null вместо файла,
пустой файл,
директория уже существует,
прерванная загрузка файла

CookChicker committed:
Merge branch 'dev' into add-rest-file_storage

CookChicker committed:
.

CookChicker committed:
Update application.properties

CookChicker committed:
Merge branch 'dev' into add-rest-file_storage
